scala/private/rules/scala_library.bzl

```diff
@@ -195,14 +133,23 @@ def scala_library_suite(
 ##
 
 def _scala_library_for_plugin_bootstrapping_impl(ctx):
-    scalac_provider = get_scalac_provider(ctx)
-    return _lib(
+    return run_phases(
         ctx,
-        scalac_provider.default_classpath,
-        True,
-        unused_dependency_checker_ignored_targets = [],
-        unused_dependency_checker_mode = "off",
-    )
+        # customizable phases
+        [
+            ("scalac_provider", phase_scalac_provider),
+            ("collect_srcjars", phase_collect_srcjars),
+            ("collect_exports_jars", phase_collect_exports_jars),
```

Comment:
I see what you are saying. I did it intentionally since the input seems to not depend on intermediate result. I suppose I can group it into `phase_init`. As we are breaking up `phase_init`, there is no good reason to change the order. I will put it back.

Reply:
Fixed.